### PR TITLE
Fix issues with MPI termination

### PR DESCRIPTION
**Pull request by latug0**
MPI_Finalize() requires that all processes agree to finish simultanously. MPI_Abort() can terminate the program as soon as one MPI Process decides it. 

Unless there is a certainty that all the MPI processes want to finish collectively (with the finalize() method), one should prefer the abort() method for exiting.

### CODE REVIEW

File: include/MFEMMGIS/Config.hxx [hunk 1]
```diff
@@ -59,6 +59,10 @@ namespace mfem_mgis {
    * This call is optional if the code exits normally.
    */
   MFEM_MGIS_EXPORT void finalize();
+  /*!
+   * \brief function that must be called if one MPI process detect an fatal error.
+   */
+  MFEM_MGIS_EXPORT void abort(int error = 1);
```
Comment (thelfer) [inline include/MFEMMGIS/Config.hxx:65]: 1. You shall declare `abort` with the `[[noreturn]]` attribute
2. Better use `EXIT_FAILURE` as de default return value

File: src/Config.cxx [hunk 2]
```diff
@@ -44,15 +45,27 @@ namespace mfem_mgis {
...
 #ifdef MFEM_USE_MPI
       MPI_Finalize();
-      this->hasFinalizeBeenCalled = true;
+      this-> ExistingOnTheWay = true;
 #endif /* MFEM_USE_MPI */
     }
   }  // end of finalize
 
-  Finalizer::~Finalizer() { this->finalize(); }
+  void Finalizer::abort(int error) {
```
Comment (thelfer) [inline src/Config.cxx:56]: Declare this method `[[noreturn]]`

**Comment by thelfer** [comment]
Just made a few comments. Once taken into account, please merge and close the PR !

**Comment by latug0** [comment]
Thanks for the comments.

**Pull request merged**